src/Ship.cpp

```diff
@@ -1135,24 +1150,34 @@ void Ship::UpdateAlertState()
 			SetAlertState(ALERT_SHIP_FIRING);
 			changed = true;
 		}
+		if (missile_detected) {
+			m_lastFiringAlert = Pi::game->GetTime();
+			SetAlertState(ALERT_MISSILE_DETECTED);
+			changed = true;
+		}
 		break;
 
 	case ALERT_SHIP_NEARBY:
-		if (!ship_is_near) {
+		if (!ship_is_near && !missile_detected) {
```

Comment:
I haven't really devoured this code, but just from a symmetry argument, why don't we need the `!ship_is_firing`, but the `!missile_detected` is needed in the condition?

Reply:
If the enemy leaves the 100km radius and the missile is still aiming for the player, the alert state will switch to "ALERT_NONE".